src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -553,16 +553,16 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
         if (class_exists('Symfony\Component\Security\Core\Exception\AuthenticationException')) {
             $r = new \ReflectionClass('Symfony\Component\Security\Core\Exception\AuthenticationException');
 
-            // with Symfony 2.4, the Security component was split into several subpackages
-            // and the translations have been moved to the symfony/security-core package
-            $translationsDir = dirname($r->getFilename()).'/../Resources/translations';
-
-            if (!file_exists($translationsDir) && file_exists($dir = dirname($r->getFilename()).'/../../Resources/translations')) {
+            if (file_exists($translationsDir = dirname($r->getFilename()).'/../../Resources/translations')) {
                 // in Symfony 2.3, translations are located in the symfony/security package
-                $translationsDir = $dir;
+                $dirs[] = $translationsDir;
             }
 
-            $dirs[] = $translationsDir;
+            if (file_exists($translationsDir = dirname($r->getFilename()).'/../Resources/translations')) {
+                // with Symfony 2.4, the Security component was split into several subpackages
+                // and the translations have been moved to the symfony/security-core package
```

Comment:
Well, we have them outside of Core for BC reasons (using them is deprecated and they are removed in 3.0)